[1.3] ci fixes (ssh-keygen and criu version bumps for almalinux 8 and fedora)

kolyshkin opened the pull request:
This is a backport of
 - #4670
 - #4728
 - #4736

to release-1.3 branch. Original description follows.

----

(from #4670)

High level overview:

- bump shfmt and shellcheck;
- get rid of some "shellcheck disable" annotations;
- bump bats to v0.11.0 so we use the same version everywhere.

----

(from #4728) 

We are seeing a ton on flakes on almalinux-8 CI job, all caused by criu inability to freeze a cgroup. This was worked around in criu (Freeze fixes and v1 kludges checkpoint-restore/criu#2545), but obviously we can't rely on a distro vendor to update the package.
Let's use a copr (thanks to @adrianreber!)

Fixes: #4273

ssh-keygen stopped working in AlmaLinux 8, fix this as well (see commit for details).

Fixes: #4731

----

(from #4736) 

Package criu-4.1-1 has a known bug [1] which is fixed in criu-4.1-2 [2],
which is currently only available in updates-testing. Add a kludge to
install newer criu if necessary to fix CI.
    
This will not be needed in ~2 weeks once the new package is promoted to
updates. 

[1]: https://github.com/checkpoint-restore/criu/issues/2650
[2]: https://bodhi.fedoraproject.org/updates/FEDORA-2025-d374d8ce17

rata commented:
Fedora is still failing related to CRIU: `not ok 39 checkpoint and restore in external network namespace`

kolyshkin commented:
> Fedora is still failing related to CRIU: `not ok 39 checkpoint and restore in external network namespace`. I guess that's why it is still a draft?

Fedora is fixed separately in https://github.com/opencontainers/runc/pull/4736 (which is also need to be backported, maybe I'll shove it in here).

kolyshkin commented:
> > Fedora is still failing related to CRIU: `not ok 39 checkpoint and restore in external network namespace`. I guess that's why it is still a draft?
> 
> Fedora is fixed separately in #4736 (which is also need to be backported, maybe I'll shove it in here).

Sorry for the confusion. I already did that :) just need to restart CI (hoping that Fedora mirrors are now updated)

kolyshkin commented:
> > Fedora is fixed separately in #4736 (which is also need to be backported, maybe I'll shove it in here).
> 
> Sorry for the confusion. I already did that :) just need to restart CI (hoping that Fedora mirrors are now updated)

I need more coffee but can't have any as it's almost 8pm here. Of course I haven't added it.

Added now, fingers crossed.

kolyshkin commented:
> > > Fedora is still failing related to CRIU: `not ok 39 checkpoint and restore in external network namespace`. I guess that's why it is still a draft?
> > 
> > 
> > Fedora is fixed separately in #4736 (which is also need to be backported, maybe I'll shove it in here).
> 
> Sorry for the confusion. I already did that :) just need to restart CI (hoping that Fedora mirrors are now updated)

No I did not. Adding now.

kolyshkin commented:
Fedora is fixed now.
